### PR TITLE
Optimizing Python + Docker deploys using Pants

### LINKED ISSUE #6: Optimize Python + Docker deployment
Use https://blog.pantsbuild.org/optimizing-python-docker-deploys-using-pants/ as source of inspiration.

Depends #4 , Depends #5

### PR DESCRIPTION
**Pull request by esirK**
## Description

This PR uses [Optimizing Python + Docker deploys using Pants](https://blog.pantsbuild.org/optimizing-python-docker-deploys-using-pants/) article to optimize our use of pants to build Docker images.

The PR also standardize use of environment variables by prefixing each one with `TWOOPSTRACKER_`, remove the mixing of `twoops-tracker`, `twoops_tracker`, etc. to just **`twoopstracker`** for package name.

Fixes #6 

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Chore (non-breaking change which does not add visible functionality but improves code quality)
- [x] This change requires a documentation update

## Screenshots

N/A

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽

If we’re moving away from `pex`, should we move away from the whole `dj` app?

**Review by thepsalmist — COMMENTED** [review]
Looks good to me @esirK

**Review by thepsalmist — COMMENTED** [review]
Looks good

File: twoops_tracker/.env.template [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
+# TwoopsTracker. If a variable is not declared here, docker-compose
+# will not pass it to container at runtime.
+# 
+# Do not assign values here. This *MUST* remain a just a list.
+# Do not do even empty assignment e.g.
+# 
+# TWOOPSTRACKER_DATABASE_URL=
+# 
+# If you do the above, TWOOPSTRACKER_DATABASE_URL will be set to empty
+# string even if a different --env-file is passed to docker-compose
```
Comment (kilemensi) [inline twoops_tracker/.env.template:11]:
```suggestion
# This is the exhaustive list of all environment variables used by
# TwoopsTracker. If a variable is not declared here, docker-compose
# will not pass it to the container at runtime.
# 
# Do not assign values here. This *MUST* remain just a list.
# Do not even do empty assignments e.g.
# 
# TWOOPSTRACKER_DATABASE_URL=
# 
# If you do the above, TWOOPSTRACKER_DATABASE_URL will be set to empty
# string even when a different --env-file is passed to docker-compose
```

File: docker-compose.yml [hunk 2]
```diff
@@ -24,30 +24,30 @@ services:
...
+    image: codeforafrica/twoopstracker:latest
     depends_on:
       - db
       - redis
-    environment:
-      - TWOOPSTRACKER_DATABASE_URL=postgres://${PGUSER:-twoops_tracker}:${PGPASSWORD:-twoops_tracker}@db:5432/${PGDATABASE:-twoops_tracker}
-      - TWOOPSTRACKER_CELERY_BROKER_URL=${TWOOPSTRACKER_CELERY_BROKER_URL:-redis://redis:6379/0}
     env_file:
-      - ./twoops_tracker/.env
+      - ./twoops_tracker/.env.template
```
Comment (esirK) [inline docker-compose.yml:33]: The code looks good but I'm not sure about this section. When performing the docker compose up, will the users have to specify the `env_file` manually? Given that we mention the `cp  twoops_tracker/.env.template twoops_tracker/.env` in the readme, shouldn't we just use `twoops_tracker/.env` as the `env_file`?

Comment (kilemensi) [reply docker-compose.yml:33]: Not a bad idea & I think that's what was originally there. 2 reasons for this switch @esirK :

1. `twoops_tracker/.env` doesn't exist in the repo. This means if we'll be hooking up to a non-existing file which I think isn't very good/robust i.e. broken by design.
2. `docker compose` offers more than [one](https://docs.docker.com/compose/environment-variables/) [way](https://docs.docker.com/compose/envvars-precedence/) to set environment variables so while we suggest (recommend?) using `.env`, it should be possible to use any of the other methods as well without breaking anything.

Comment (esirK) [reply docker-compose.yml:33]: Makes sense.

**Comment by esirK** [comment]
I think this is good to go now @kilemensi  @thepsalmist

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**